middleware/auth.py

```diff
@@ -0,0 +1,75 @@
+import re
+import json
+import yaml
+
+from fastapi import Depends, HTTPException, status
+from fastapi.security import HTTPBearer, HTTPAuthorizationCredentials
+from yaml import SafeLoader
+from cryptography.fernet import Fernet
+
+from app.config import iRODSConfig
+
+security = HTTPBearer()
+
+
+class Authenticator:
+    def __init__(self):
+        self.key = Fernet.generate_key()
+        self.fernet = Fernet(self.key)
+        self.access_token = None
+        self.authorized_email = None
+        self.public_access = False
+        self.current_user = None
```

Comment:
Can this class work with multiple users at the same time?

Reply:
I have thought about this question. Haven't tested yet, probably not support. Will take a look at this.